### ISSUE TITLE
Enhance Pagefind to support search with non-matching text

### ISSUE THREAD

**Issue by kenmorse** [CloudCannon/pagefind#445]
From [discussion](https://github.com/CloudCannon/pagefind/discussions/150).

When people enter more free-form text in an input field or textarea, Pagefind won't always return results. For example, enter this in the search on the [Pagefind page](https://pagefind.app/):

    I need help with pagefind and hugo along with jekyll

and you'll see 

    No results for I need help with pagefind and hugo along with jekyll

Even though many of those words are indeed on the site, no results are returned, even with the new 1.0.x versions.

It would be great to see Pagefind support free-form searches like this.

**Comment by bglw** [comment]
Agreed! I didn't get a chance to tackle this prior to 1.0, but I do have some ideas so should be able to look at this in an upcoming release 🙂

**Comment by kenmorse** [comment]
Happy to help in any way I can, including testing, @bglw

**Comment by kenmorse** [comment]
@bglw – does the [Customize ranking](https://pagefind.app/docs/ranking/) work you did for version [1.1.0](https://github.com/CloudCannon/pagefind/releases/tag/v1.1.0) help with this at all?

**Comment by bglw** [comment]
Hmm, not immediately, no. But it might pose as an interesting place to add a setting around this 🤔

**Comment by kenmorse** [comment]
I thought maybe the new [Playground feature](https://pagefind.app/docs/playground/) would help identify why a more free-form search doesn't work, but using the same sample sentence as in the op produces no results. Would have thought that missing keywords would have a low score but they don't show up; maybe some early processing going on that prevents results? 

<img width="968" height="793" alt="Image" src="https://github.com/user-attachments/assets/f7700fcc-4eaf-4763-bd50-d1d09cda997b" />

**Comment by bglw** [comment]
Note: A lot of ranking work has gone into 1.5.0, but conversational sentence searches still need some attention.

**Comment by kenmorse** [comment]
Completely agree, @bglw — some great stuff in 1.5.0 and conversational sentence searches will make pagefind even better.
